add home portal entity script and make it go to correct welcome

howard-stearns opened the pull request:
fixes https://highfidelity.fogbugz.com/f/cases/2042/protocol-mismatch-in-my-own-home

howard-stearns commented:
This adds the ftue/home portal script to the unpublished-scripts tree in the source. It modifies it only in that it recognizes either an empty destination string or the existing welcome/mumble string as meaning that we should go to entry.  (Going to entry is handled by Interface as going to the correct welcome area for the current build, rather than to a script-coded location.)

Separate from this check-in, we still have to replace the existing portal.js in the home content set with this new version.

howard-stearns commented on unpublishedScripts/DomainContent/Home/portal.js:
```diff
@@ -0,0 +1,48 @@
+            if (portalDestination[0] == '/') {
+                print("Teleporting to " + portalDestination);
+                Window.location = portalDestination;
+            } else if (portalDestination.indexOf('welcome/3.18552,0.164796,-0.300168/0,0.763363,0,0.64597') >= 0) {
```
I'm confused. I thought:
* As I mention above, separate from this check-in, we still hgave to replace the existing portal.js in the home content set.
* No one in beta has the new content yet (other than people using dev-download, see below).
* We are preparing a new content set that will go out with the new beta and steam download. That new set will come with atp content, including whatever is the current version of this script at the time.
* Even people who have been using dev-downlad are now given the option to update their content whenever the content changes. This is a content change.

howard-stearns replied:
Line 30 recognizes the existing userData string, so that we don't HAVE to change it at the same time as changing the script asset. But line 28 also recognizes empty userData, so we have the option of changing the userData also.

I have no objection to removing the line 30 part and changing the userData at the same time. Just trying to keep the level of coordination down. (There are plenty of other changes I'd like to make, but I'm trying to keep to the minimum given where we are in the release cycle.)

huffman replied:
I see - I'd say let's just remove the backwards compatibility. I'll make sure to update the entity when I update the script.

hifi-gustavo commented:
Build console output for commit: 4c4ec37eb300a6283850f992bbf30efff0d271fb available at: https://gist.github.com/e73e0e74ac3e36cc4688850e70c29fff<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8890-4c4ec37.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8890-4c4ec37.dmg

hifi-gustavo commented:
Build console output for commit: 963fd68cd9c39cfda2cae16b7578716adda158f5 available at: https://gist.github.com/e73e0e74ac3e36cc4688850e70c29fff<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8890-963fd68.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8890-963fd68.dmg

cozza13 commented:
ok to test

hifi-gustavo commented:
Build console output for commit: 963fd68cd9c39cfda2cae16b7578716adda158f5 available at: https://gist.github.com/e73e0e74ac3e36cc4688850e70c29fff<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8890-963fd68.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8890-963fd68.dmg

howard-stearns commented:
tested by me in localhost:
- removed portal.js from asset manager. re-added from this source.
- removed portal-to-welcome's userData (but left the other /tutorial portal alone)
- tested both portals.
